tests/test_spatial_tools.py

```diff
@@ -178,6 +208,136 @@ def make_gdal_hillshade(filepath) -> np.ndarray:
     # Make sure that this doesn't create weird division warnings.
     xdem.spatial_tools.hillshade(dem.data, dem.res)
 
+class TestRobustFitting:
+
+    @pytest.mark.parametrize("pkg_estimator", [('sklearn','Linear'), ('scipy','Linear'), ('sklearn','Theil-Sen'),
+                                           ('sklearn','RANSAC'),('sklearn','Huber')])
+    def test_robust_polynomial_fit(self, pkg_estimator: str) -> None:
+
+        np.random.seed(42)
+
+        # x vector
+        x = np.linspace(1, 10, 1000)
+        # exact polynomial
+        true_coefs = [-100, 5, 3, 2]
+        y = true_coefs[0] + true_coefs[1] * x + true_coefs[2] * x ** 2 + true_coefs[3] * x ** 3
+
+        coefs, deg = xdem.spatial_tools.robust_polynomial_fit(x, y, linear_pkg=pkg_estimator[0], estimator=pkg_estimator[1], random_state=42)
+
+        assert deg == 3 or deg == 4
+        assert np.abs(coefs[0] - true_coefs[0]) <= 100
```

Comment:
Why do you allow such a large error? Should it not be much closer to the actual value than that??

Reply:
Slight variations in high-order coefficients result in good fits with very different low-order coefficients, hence the variability...
Don't know if there is a smartest way to check a polynomial accounting for this effect